fbw-a380x/src/wasm/systems/a380_systems/src/fuel/fuel_quantity_management_system.rs

```diff
@@ -369,20 +364,21 @@ impl RefuelApplication {
         let zero_fuel_weight_cg_rounded = (zero_fuel_weight_cg_percent_mac).floor() as u32;
 
         let mut trim_fuel = Mass::default();
+        let target_zfw = self.trim_tank_map.params.target_zfw.clone();
+        let trim_tank_targets = self.trim_tank_map.trim_tank_targets.tables.clone();
 
-        for (index, range) in self.trim_tank_map.params.target_zfw.iter().enumerate() {
+        for (range, zfw_category) in target_zfw.iter().zip(&trim_tank_targets) {
             if zero_fuel_weight >= Mass::new::<kilogram>(range.start as f64)
                 && zero_fuel_weight <= Mass::new::<kilogram>(range.end as f64)
             {
                 trim_fuel = self.lookup_trim_fuel_from_mapping(
-                    index,
+                    zfw_category,
                     total_desired_fuel_rounded,
                     zero_fuel_weight_cg_rounded,
                 );
                 break;
             }
         }
-
         trim_fuel
     }
```

Comment:
```suggestion
        for (range, zfw_category) in target_zfw.iter().zip(trim_tank_targets) {
            if (range.start as f64..=range.end as f64).contains(&zero_fuel_weight.get::<kilogram>())
            {
                return Self::lookup_trim_fuel_from_mapping(
                    &self.trim_tank_map.params.target_zfw_cg,
                    zfw_category,
                    total_desired_fuel_rounded,
                    zero_fuel_weight_cg_rounded,
                );
            }
        }

        Mass::default()
```
Some more improvements. You could make `ZfwRange` an alias to a `RangeInclusive<u32>` too if you wanted.